### PR TITLE
Rwa 1835

### PR DESCRIPTION
**Pull request by chrisdavecm**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-1835

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/entities/TaskHistoryResource.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@ToString
+@Getter
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
+@Entity(name = "taskHistory")
+@SuppressWarnings({"PMD.TooManyFields"})
+public class TaskHistoryResource {
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/entities/TaskHistoryResource.java:19]: Not important but could this be a Java 17 Record?  There's some customisation required for equalandhashcode so may be more trouble than it's worth.

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/response/TaskHistoryResponse.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.controllers.response;
+
+import lombok.Builder;
+import lombok.Data;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.entities.TaskHistoryResource;
+
+import java.util.List;
+
+@Builder(toBuilder = true)
+@Data
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/response/TaskHistoryResponse.java:10]: As above we don't need to use this any more with `Record`s in Java 17.  Don't need to fix here if easier.

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/schedulers/LogicalReplicationCreatorScheduler.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+public class LogicalReplicationCreatorScheduler {
+    private static final org.slf4j.Logger LOGGER = LoggerFactory.getLogger(LogicalReplicationCreatorScheduler.class);
+
+    @Autowired
+    private MIReportingService miReportingService;
+
+    @Scheduled(fixedDelay = 1000)
+    public void scheduled() {
+        LOGGER.debug("Postgresql logical replication scheduler executed");
+        miReportingService.logicalReplicationCheck();
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/schedulers/LogicalReplicationCreatorScheduler.java:22]: Will this check every second even if the previous check has blocked for a few seconds?
I am just thinking about what happens in that case?  Could you get 2 attempts to create a publication?  Does that matter or does it just work?

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReportingService.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    }
+
+    public List<TaskHistoryResource> findByTaskId(String taskId) {
+        return taskHistoryRepository.getByTaskId(taskId);
+    }
+
+    public void logicalReplicationCheck() {
+        LOGGER.debug("Postgresql logical replication check executed");
+        if (isReplicationSlotPresent()) {
+            if (!isPublicationPresent()) {
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReportingService.java:59]: As the comment above.  Could you get multiple threads attempting to do this at the same time?  Will this have unwanted side effects?

**Review by raghera — APPROVED** [review]
Since this is an inactive version of the changes I have approved as long as the comments are considered in the active version later.

The comments are mainly questions so should not block this PR from moving ahead.

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/schedulers/LogicalReplicationCreatorScheduler.java (hunk 3 above)
Comment (chrisdavecm) [reply src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/schedulers/LogicalReplicationCreatorScheduler.java:24]: Should be transactional

Comment (chrisdavecm) [reply src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/schedulers/LogicalReplicationCreatorScheduler.java:24]: Have run it on perftest and no problems

**Review by rasikasri — APPROVED** [review]

**Pull request merged**